pydm/widgets/base.py

```diff
@@ -1403,6 +1410,39 @@ def eventFilter(self, obj, event):
 
         return PyDMWidget.eventFilter(self, obj, event)
 
+    @Property(str)
+    def channel(self) -> Optional[str]:
+        """
+        The channel address in use for this widget.
+
+        Returns
+        -------
+        channel : str
+            Channel address
+        """
+        if self._channel:
+            return str(self._channel)
+        return None
+
+    @channel.setter
+    def channel(self, value: str) -> None:
+        """
+        The channel address in use for this widget. Also sets up a monitor on the DISP field.
+        """
+        if self._channel != value:
+            self.set_channel(value)
+            if value is not None and self._disp_channel != f'{value}.DISP':
+                if self._disp_channel is not None:
+                    self._disp_channel.disconnect()
+                self._disp_channel = f'{value}.DISP'
```

Comment:
A few thoughts:
1. An EPICS record-specific thing like this seems like it should only apply to certain protocols (CA, PVA)
2. PV names may already include a field name: this could end up being `PV:NAME.VAL.DISP`, couldn't it?
3. Similarly, PV names may include a [channel filter](https://epics.anl.gov/base/R7-0/4-docs/filters.html) as in e.g.  `PV:NAME.[3:5]` - so stripping off the field/filter will be necessary. Fortunately both are guaranteed to be preceded by a `.`, so `channel.split(".", 1)[0] if "." in channel else channel` can be sufficient
4. Python IOCs may not have fields at all depending on how they are configured. I assume if this doesn't connect it's just a no-op, right? So then we just waste a search for each PV of that type.
5. I didn't realize disable put didn't manifest itself as showing "read-only" in `cainfo` - that's new and confusing to me. I assume that's where you started and this solution came about because of that?